docker_environment/Dockerfile

```
@@ -59,7 +60,11 @@ RUN ln -sfn /usr/bin/python3.6 /usr/bin/python & \
     ln -s /usr/bin/pip3 /usr/bin/pip
 
 # Upgrade pip
-RUN pip install --upgrade pip
+RUN pip3 install --upgrade pip
+
+# Install poetry
+RUN pip3 install poetry && poetry config virtualenvs.create false
```

I'd suggest we use a fixed `poetry` version.

Also, I'd suggest we stick to the installation instructions recommended by `poetry`, that can be found here: https://python-poetry.org/docs/#installation

> Poetry provides a custom installer that will install poetry isolated from the rest of your system by vendorizing its dependencies. This is the recommended way of installing poetry.

Since installing `poetry` with `pip` may cause conflicts, as they mention here: https://python-poetry.org/docs/#installing-with-pip

> Be aware that it will also install Poetry's dependencies which might cause conflicts with other packages.

So, I think the installation may be as follows:

```suggestion
RUN set -x \
	&& apt-get install -y curl \
	&& curl -sSL  https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py -o get-poetry.py \
	&& python3 get-poetry.py --version 1.1.4 \
	&& poetry config virtualenvs.create false \
	&& rm -rf get-poetry.py
```

---

Done